File: 6th Semester/Programming Paradigms/NP2 Exercises/ex3.py
# Escreva um programa que leia três números e que imprima o maior e o menor

def ler_inteiro(mensagem: str) -> int:
    while True:
        output = str(input(mensagem)).strip()
        if (output.replace('-','').isnumeric()):
            return int(output)
        print('Por favor insira um número inteiro válido.')

def main():
    print("Programa compara números (inteiros)")
    num1 = ler_inteiro('Digite o número 1: ')
    num2 = ler_inteiro('Digite o número 2: ')
    num3 = ler_inteiro('Digite o número 3: ')
    maior = max([num1, num2, num3])
    menor = min([num1, num2, num3])
    if (num1 == num2 == num3):
        print(f"Não tem menor ou maior, todos os números são {num1} iguais.")
    else:
        print(f"O menor número é {menor} e o maior é {maior}")
    

if __name__ == "__main__":
    main()

File: 6th Semester/Programming Paradigms/NP2 Exercises/ex3.c
// Escreva um programa que leia três números e que imprima o maior e o menor
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define TAMANHO_OUTPUT 100
#define QUANTIDADE_NUMS 3

int lerInteiro(char mensagem[]) {
    char output[TAMANHO_OUTPUT];
    int inteiro;
    while(1) {
        printf("%s", mensagem);
        fgets(output, TAMANHO_OUTPUT, stdin);
        if (!((inteiro = atoi(output)) == 0 && output[0] != '0'))
            return inteiro;
        printf("Por favor insira um número inteiro válido!\n");
    }
}

int main() {
    printf("Programa compara números (inteiros)\n");
    int maior, menor, numInteiro, diferentes;
    char mensagem[40];
    for (int i = 0; i < QUANTIDADE_NUMS; i++) {
        sprintf(mensagem, "Digite o número %i: ", i+1);
        numInteiro = lerInteiro(mensagem);
        if (i == 0) {
            maior = numInteiro;
            menor = numInteiro;
            diferentes = 0;
        }
        if (maior > numInteiro) {
            maior = numInteiro;
            diferentes = 1;
        }
        else if (menor < numInteiro) {
            maior = numInteiro;
            diferentes = 1;
        }
    }
    if (!diferentes) printf("Não tem menor ou maior, todos os números são %d iguais.\n", numInteiro);
    else {
        printf("O menor número é %d e o maior é %d.\n", menor, maior);
    }
    return 0;
}